common/src/main/java/com/wynntils/wynn/model/quests/DialogueHistoryQueries.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.quests;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.wynn.model.container.ScriptedContainerQuery;
+import com.wynntils.wynn.utils.InventoryUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.world.item.ItemStack;
+
+public class DialogueHistoryQueries {
+    private static final Pattern DIALOGUE_HISTORY_PAGE_PATTERN = Pattern.compile("§7Page \\[(\\d+)/(\\d+)\\]");
+
+    private List<List<String>> newDialogueHistory;
+
+    protected void scanDialogueHistory() {
+        ScriptedContainerQuery.QueryBuilder queryBuilder = ScriptedContainerQuery.builder(
+                        "Quest Book Dialogue History Query")
+                .onError(msg -> WynntilsMod.warn("Problem getting dialogue history in Quest Book: " + msg))
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(QuestManager.getQuestBookTitle(1))
+                .processContainer((c) -> {
+                    ItemStack dialogueHistoryItem = c.items().get(0);
+
+                    if (!ComponentUtils.getCoded(dialogueHistoryItem.getHoverName())
+                            .equals("§bDialogue History")) return;
+
+                    for (String line : ItemUtils.getLore(dialogueHistoryItem)) {
+                        Matcher matcher = DIALOGUE_HISTORY_PAGE_PATTERN.matcher(line);
+
+                        if (matcher.matches()) {
+                            int pageCount = Integer.parseInt(matcher.group(2));
+                            createActualDialogueHistoryQuery(pageCount);
+                            break;
+                        }
+                    }
+                });
+
+        queryBuilder.build().executeQuery();
+    }
+
+    private void createActualDialogueHistoryQuery(int pageCount) {
```

Comment:
I would suggest removing the `actual` from this, as this is more of a continuation of the query

Reply:
Well, it *is* creating the actual query, since the other part is just a helper to extract the number of pages we need to query.

I've renamed it slightly, and added some more clarification, so the flow is hopefully clearer.

Me and boylai need to come up with a better way to create dynamic queries, so we don't need to know the number of pages beforehand. But that is a future exercise.